gammapy/visualization/utils.py

```diff
@@ -341,8 +341,8 @@ def plot_distribution(
         )
         cells_in_grid = rows * cols
     else:
-        axes = ax
-        cells_in_grid = len(ax.flatten())
+        axes = np.array([ax])
+        cells_in_grid = axes.size
 
     if not isinstance(axes, np.ndarray):
         axes = np.array([axes])
```

Comment:
```suggestion
        axes = np.array([axes])
    cells_in_grid = axes.size
```

Reply:
In this case, cells in grid is initialized over a previous instance if ax=None (see line 342). Should we just change this as well?

Reply:
I see what you mean, let me redo my comments then!